### PR TITLE
Add thrift support for parquet column and offset indexes

### PR DESCRIPTION
**Pull request by etseidl**
Adds some necessary structs to parquet.hpp as well as methods to CompactProtocolReader/Writer to address #9268

I can add tests if necessary once #11177 is merged, or testing can be deferred to be included in a future PR (based on #11171)

### CODE REVIEW

**Comment by GPUtester** [comment]
Can one of the admins verify this patch?

File: cpp/src/io/parquet/compact_protocol_writer.hpp [hunk 1]
```diff
@@ -50,6 +50,8 @@ class CompactProtocolWriter {
   size_t write(const KeyValue&);
   size_t write(const ColumnChunk&);
   size_t write(const ColumnChunkMetaData&);
+  size_t write(const PageLocation&);
+  size_t write(const OffsetIndex&);
```
Comment (devavret) [inline cpp/src/io/parquet/compact_protocol_writer.hpp:54]: No write support for ColumnIndex?

Comment (etseidl) [reply cpp/src/io/parquet/compact_protocol_writer.hpp:54]: ColumnIndex is encoded on the device, so I didn't add it to the writer

Comment (devavret) [reply cpp/src/io/parquet/compact_protocol_writer.hpp:54]: oh just like page headers

**Review by devavret — APPROVED** [review]

**Comment by PointKernel** [comment]
ok to test

**Review by PointKernel — APPROVED** [review]
LGTM

**Comment by PointKernel** [comment]
@gpucibot merge

**Pull request merged**